Source/ARTAuthOptions.m

```diff
@@ -135,4 +135,52 @@ - (BOOL)isMethodGET {
     return [_authMethod isEqualToString:@"GET"];
 }
 
+- (ARTAuthOptions *)replaceWith:(ARTAuthOptions *)options {
```

Comment:
Why do we even need to do this? Why not just use the new `options` instead of cloning the old one and copying everything from the new one into it? Unless I'm missing something, that just gives you a clone of the new one that you could've gotten with `[options copy]` in case you really need a copy.

Plus, it's full of unsafe casts from ARTAuthOptions to ARTClientOptions, and I don't see the point of the synchronized block given that the whole library is single-threaded.